services/QuillLMS/client/app/bundles/PremiumHub/containers/diagnosticGrowthReports/index.tsx

```diff
@@ -28,17 +28,78 @@ export const DiagnosticGrowthReportsContainer = ({
 }) => {
 
   const [activeTab, setActiveTab] = React.useState<string>(OVERVIEW)
+  const [noDiagnosticDataAvailable, setNoDiagnosticDataAvailable] = React.useState<boolean>(!!passedData)
 
   function handleTabChange(e) {
     setActiveTab(e.currentTarget.value)
   }
 
+  function handleSetNoDiagnosticDataAvailable(value: boolean) {
+    setNoDiagnosticDataAvailable(value)
+  }
+
+  const sharedProps = {
+    searchCount,
+    selectedGrades: mapItemsIfNotAll(selectedGrades, availableGrades, 'value'),
+    selectedSchoolIds: selectedSchools.map(school => school.id),
+    selectedTeacherIds: mapItemsIfNotAll(selectedTeachers, availableTeachers),
+    selectedClassroomIds: mapItemsIfNotAll(selectedClassrooms, availableClassrooms),
+    selectedTimeframe: SELECTED_TIMEFRAME,
+    pusherChannel,
+    hasAdjustedFiltersFromDefault,
+    handleSetNoDiagnosticDataAvailable,
+    passedData: null
+  }
+
+  function renderContent() {
+
+    if (noDiagnosticDataAvailable) {
+      return(
+        <div className="no-diagnostic-data-container">
+          <h2>There are not yet any completed diagnostics.</h2>
+          <p>The Quill Diagnostic enables you to provide a Pre-Diagnostic at the start of the year to identify skills students need to practice. The Post-Diagnostic allows you to then measure students&apos; learning gains over the course of the school year.</p>
+          <p>At the moment, there are no teachers connected to your admin account who have completed a diagnostic. Once at least one teacher has completed a diagnostic with at least one student, you will be able to see their results in this report. View our guide our guide to learn how teachers can assign diagnostics to their students.</p>
+          <a className="assign-link focus-on-light" href="" rel="noopener noreferrer" target="_blank">How to Assign a Diagnostic</a>
+        </div>
+      )
+    }
+
+    return(
+      <React.Fragment>
+        <div className="tabs-for-pages-container">
+          <button className={`interactive-wrapper performance-type-button overview ${activeTab === OVERVIEW ? 'active' : ''}`} onClick={handleTabChange} value={OVERVIEW}>
+            <img alt="" src={activeTab === OVERVIEW ? barChartWhiteIconSrc : barChartGreySrc} />
+            <span>Performance Overview</span>
+          </button>
+          <button className={`interactive-wrapper performance-type-button skill ${activeTab === SKILL ? 'active' : ''}`} onClick={handleTabChange} value={SKILL}>
+            <img alt="" src={activeTab === SKILL ? pencilWhiteIconSrc : pencilGreyIconSrc} />
+            <span>Performance by Skill</span>
+          </button>
+          <button className={`interactive-wrapper performance-type-button student ${activeTab === STUDENT ? 'active' : ''}`} onClick={handleTabChange} value={STUDENT}>
+            <img alt="" src={activeTab === STUDENT ? groupOfStudentsWhiteIconSrc : groupOfStudentsGreyIconSrc} />
+            <span>Performance by Student</span>
+          </button>
```

Comment:
All the ternary logic in here is effectively duplicated and feels a little messy. I think we should do something like:

```
const performanceButtons = [
  {
    tab: OVERVIEW,
    displayName: 'Performance Overview',
    inactiveIconSrc: barChartGreySrc,
    activeIconSrc: barChartWhiteIconSrc
  },
  ...
]
```

Then iterate through each of them to generate the button elements in their own function.

Reply:
love that suggestion-- I'll make this update